builtin/logical/pki/path_roles.go

```diff
@@ -272,14 +280,37 @@ func (b *backend) pathRoleDelete(
 
 func (b *backend) pathRoleRead(
 	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-	role, err := b.getRole(req.Storage, data.Get("name").(string))
+	roleName := data.Get("name").(string)
+	if roleName == "" {
+		return logical.ErrorResponse("missing role name"), nil
+	}
+
+	role, err := b.getRole(req.Storage, roleName)
 	if err != nil {
 		return nil, err
 	}
 	if role == nil {
 		return nil, nil
 	}
 
+	// Upgrade generate_lease in role
```

Comment:
This upgrade code needs to be with the rest of the upgrade code in `getRole` (will also simplify it). Otherwise it won't run unless they explicitly read the role.